compendium/DeclarativeServices/src/ComponentContextImpl.cpp

```diff
@@ -83,21 +84,36 @@ namespace cppmicroservices
                             auto& serviceMap = (*boundServicesCacheHandle)[refName];
                             if (refScope == cppmicroservices::Constants::SCOPE_BUNDLE)
                             {
-                                serviceMap.push_back(bc.GetService(sRefU));
+                                auto svc = bc.GetService(sRefU);
+                                if (svc)
+                                {
+                                    foundAtLeastOneValidBoundService = true;
+                                }
+                                serviceMap.push_back(svc);
```

Comment:
Should this line be within the scope of the 'if' block above? See line 101 below.

Reply:
No, I'm maintaining the existing behavior which was to place the service object (even if it was `nullptr`) within the `serviceMap`